kernel/src/common/utils.rs

```diff
@@ -15,23 +15,17 @@
 /// destructor.
 #[macro_export]
 macro_rules! static_init {
-    ($T:ty, $e:expr, $size:expr) => {
-        // Ideally we could use mem::size_of<$T> here instead of $size, however
-        // that is not currently possible in rust. Instead we write the size as
-        // a constant in the code and use compile-time verification to see that
-        // we got it right
+    ($T:ty, $e:expr, $size:expr) => (static_init!($T, $e));
```

Comment:
Why not get rid of all sizes entirely instead? It seems ungood to have an interface that specifies sizes which are actually unused under the hood. Moreover, doesn't this mean that nothing is even checking if the sizes are correct anymore?